com.unity.multiplayer.mlapi/Runtime/Messaging/RpcQueue/RpcQueueContainer.cs

```diff
@@ -740,14 +703,112 @@ public void Shutdown()
             ClearParameters();
         }
 
+        /// <summary>
+        /// Cleans up our instance count and warns if there instantiation issues
+        /// </summary>
+        public void Dispose()
+        {
+            Shutdown();
+
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+            if (s_RpcQueueContainerInstances > 0)
+            {
+                if (NetworkLog.CurrentLogLevel == LogLevel.Developer)
+                {
+                    NetworkLog.LogInfo($"[Instance : {s_RpcQueueContainerInstances}] {nameof(RpcQueueContainer)} disposed.");
+                }
+
+                s_RpcQueueContainerInstances--;
+            }
+            else //This should never happen...if so something else has gone very wrong.
+            {
+                if (NetworkLog.CurrentLogLevel >= LogLevel.Normal)
+                {
+                    NetworkLog.LogError($"[*** Warning ***] {nameof(RpcQueueContainer)} is being disposed twice?");
+                }
+
+                throw new Exception("[*** Warning ***] System state is not stable!  Check all references to the Dispose method!");
+            }
+#endif
+        }
+
         /// <summary>
         /// RpcQueueContainer - Constructor
+        /// Note about processExternally: this values determines if it will register with the Network Update Loop
+        /// or not.  If not, then most likely unit tests are being preformed on this class.  The default value is false.
         /// </summary>
-        /// <param name="processInternally">determines if it handles processing internally or if it will be done externally</param>
-        /// <param name="isLoopBackEnabled">turns loopback on or off (primarily debugging purposes)</param>
-        public RpcQueueContainer(bool processExternally)
+        /// <param name="maxFrameHistory"></param>
+        /// <param name="processExternally">determines if it handles processing externally</param>
+        public RpcQueueContainer(NetworkManager networkManager, uint maxFrameHistory = 0, bool processExternally = false)
         {
+            m_NetworkManager = networkManager;
```

Comment:
Can we just assert if m_networkManger is `null`, and then since it (will be) `readonly` then we don't need to check for null nor worry about getting in a weird state?

Reply:
Yep... if that happens... then something outside of the RpcQueueContainer has gone very wrong.